libtiny_tui/src/input_area/input_line.rs

```diff
@@ -2,30 +2,175 @@ use crate::{config::Colors, termbox, utils};
 use std::{cmp::min, ops::RangeBounds, vec::Drain};
 use termbox_simple::Termbox;
 
+/// Cache that stores the state of InputLine's height calculation.
+/// `line_count` is used as the dirty bit to invalidate the cache.
+#[derive(Clone, Debug)]
+struct LineDataCache {
+    /// Indices to split on when we draw, not always whitespaces
+    ws_indices: Vec<i32>,
```

Comment:
Ahh, so you also fix #202 with this, right? Let's rename this to `split_indices`.

Reply:
No, it doesn't fix that #202, sadly. That issue existed before any of the input changes. I've been thinking about how to fix that problem though.

I will rename.

Reply:
Ah, I'm completely confused. This is about the input field of course it doesn't fix #202.